src/sgb-xi/index.ts

```diff
@@ -27,29 +26,50 @@ const makeBillingFile = (
         dateiindikator,
         rechnungsart
     } = billing;
-    const absenderIK = invoices[0].leistungserbringer.absenderIK;
+    const absenderIK = absender(billing, invoices[0]).ik;
     const datenaustauschreferenz = datenaustauschreferenzJeEmpfaengerIK[empfaengerIK] || 1;
     const laufendeDatenannahmeImJahr = laufendeDatenannahmeImJahrJeEmpfaengerIK[empfaengerIK] || 1;
     const anwendungsreferenz = makeAnwendungsreferenz(kassenart, laufendeDatenannahmeImJahr, billing);
     const dateiname = makeDateiname(dateiindikator, datenaustauschreferenz - 1); // todo: resaearch „verfahrensversion”, maybe in Auftragsdatei documentation?
     let messageNumber = 0;
+    let invoiceIndex = 0;
+
+    // sort leistungen und einsätze by start date
+    invoices.map(invoice => ({
+        ...invoice,
+        faelle: invoice.faelle.map(fall => ({
+            ...fall,
+            einsaetze: fall.einsaetze.map(einsatz => ({
+                ...einsatz,
+                leistungen: einsatz.leistungen.sort((a, b) =>
+                    (a?.leistungsBeginn?.getTime() || Number.MAX_VALUE)
+                    - (b?.leistungsBeginn?.getTime() || Number.MAX_VALUE)
+                )
+            })).sort((a, b) =>
+                (a?.leistungsBeginn?.getTime() || Number.MAX_VALUE)
+                - (b?.leistungsBeginn?.getTime() || Number.MAX_VALUE)
+            )
+        }))
+    }));
```

Comment:
Find this a bit hard to read.

You are mapping the invoices, copying everything, but basically change nothing except the order of the leistungen and einsaetze. 

Questions:
1. But where are you mapping it to? It looks like you are copying the data structure, but you assign the result of the `map` to nothing
2. If you are copying the data structure and then sort some parts of it, wouldn't it be both shorter and more readable if you copied the whole thing, then sorted the leistungen and einsaetze (in a for loop)
3. Why not do the sorting on the original data structure?

Reply:
I totally agree with you, this could be way easier by sorting the original data structure. I changed this accordingly.